lib/linux_kernel/kthread.c

```diff
@@ -228,20 +228,24 @@ M0_INTERNAL int m0_thread_confine(struct m0_thread *q,
 		  put_task_struct(p);
 		*/
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0)
+#ifdef CONFIG_X86_64
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0)
 		cpumask_copy(&p->cpus_mask, cpuset);
-#else
+	#else
 		cpumask_copy(&p->cpus_allowed, cpuset);
-#endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
+	#endif
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
 		p->nr_cpus_allowed = nr_allowed;
-#else
+	#else
 		p->rt.nr_cpus_allowed = nr_allowed;
+	#endif
+#else /*aarch64*/
+	cpumask_copy(&p->cpus_allowed, cpuset);
+	p->nr_cpus_allowed = nr_allowed;
 #endif
-
-		/* cause current task to migrate immediately by blocking */
-		if (p == current && !cpumask_test_cpu(task_cpu(p), cpuset))
-			schedule_timeout_uninterruptible(1);
+	/* cause current task to migrate immediately by blocking */
+	if (p == current && !cpumask_test_cpu(task_cpu(p), cpuset))
+		schedule_timeout_uninterruptible(1);
```

Comment:
Why?

Reply:
This to align the code as per it depth

Reply:
But the change of this indentation is wrong.

The original code is the correct one.
And the code 
```
#elif defined CONFIG_AARCH64 /*aarch64*/
	        cpumask_copy(&p->cpus_allowed, cpuset);
	        p->nr_cpus_allowed = nr_allowed;
#endif
```
should have one more depth.

Reply:
framed the code  by putting with  one more tab

Reply:
```
	/* cause current task to migrate immediately by blocking */
	if (p == current && !cpumask_test_cpu(task_cpu(p), cpuset))
		schedule_timeout_uninterruptible(1);
```
One more tab is needed for this piece of code.